src/main/java/com/glodblock/github/nei/recipes/FluidRecipe.java

```diff
@@ -24,13 +25,29 @@ public static void addRecipeMap(String recipeIdentifier, IRecipeExtractor extrac
         }
     }
 
-    public static List<OrderStack<?>> getPackageInputs(IRecipeHandler recipe, int index) {
+    public static List<OrderStack<?>> getPackageInputs(IRecipeHandler recipe, int index, boolean priority) {
         TemplateRecipeHandler tRecipe = (TemplateRecipeHandler) recipe;
         if (tRecipe == null || !IdentifierMap.containsKey(tRecipe.getOverlayIdentifier())) return new ArrayList<>();
         if (tRecipe.getOverlayIdentifier() == null) return getPackageInputsLegacy(recipe, index);
         IRecipeExtractor extractor = IdentifierMap.get(tRecipe.getOverlayIdentifier());
         if (extractor == null) return new ArrayList<>();
         List<PositionedStack> tmp = new ArrayList<>(tRecipe.getIngredientStacks(index));
+        List<PositionedStack> fluidClone = new ArrayList<>();
+        if (priority) {
+                for (int i = 0; i < tmp.size(); i++) {
+                    if(tmp.get(i).item.getItem() instanceof GT_FluidDisplayItem) {
+                        PositionedStack Stack = tmp.get(i);
+                        fluidClone.add(Stack);
+                        }
+                    }
+            for (int i = 0; i < fluidClone.size(); i++) {
+                tmp.remove(fluidClone.get(i));
+                }
+            Collections.reverse(fluidClone);
+            for (int i = 0; i < fluidClone.size(); i++) {
+                tmp.add(0, fluidClone.get(i));
+            }
+        }
         return extractor.getInputIngredients(tmp, recipe, index);
```

Comment:
i will suggest you modify the `List<OrderStack<?>>` from `extractor.getInputIngredients(tmp, recipe, index)` instead of messing with `List<PositionedStack>`. otherwise it will only work with gt recipes